meetnearme/api
# Local DyanmoDB Setup with Docker Compose

# Description 

The PR adds a Docker container to run the local implementation of DynamoDB. It aims to integrate a baseline dynamodb running locally and seeds the DB with basic event data. Also, it includes the API interface in Go to list all events in the Events table and to add new individual events. The changes mainly consist of moving from the mock user to using events as the data. 



## Type of Change



# Testing 

1. Pull changes into local branch
2. Make sure the Docker daemon is running (Docker Desktop)
3. run `$ docker compose build` from root directory
4. run `$ docker compose up` from root directory 
    - Screenshots below showing expected console output from successful run 
5. In another terminal window run `curl localhost:3001/event` - wait for response text from lambda to list slice of events
6. Test Insert new event 

`curl -X POST -H 'Content-Type: application/json' -d '{"name": "Chess Tournament", "description": "Join the junior chess tournament to test your abilities", "datetime": "2024-03-13T15:07:00", "address": "15 Chess Street", "zip_code": "84322", "country": "USA"}' http://0.0.0.0:3001/event` 

7. Check insertion into list
`curl localhost:3001/event`
